### PR TITLE
Stricter Markdown testing

### PR DESCRIPTION
**Pull request by krlmlr**
with the new `expect_equivalent_rd()` helper.

Had to adapt/fix a few tests, too.

### CODE REVIEW

File: tests/testthat/test-rd-markdown.R [hunk 1]
```diff
@@ -258,8 +254,6 @@ test_that("markdown is parsed in all fields where it is supported", {
     #'
     #' @seealso See also **with bold**
     #'
-    #' @keywords Keywords **with bold**
-    #'
```
Comment (gaborcsardi) [inline tests/testthat/test-rd-markdown.R]: What's wrong with the keywords?

Comment (hadley) [inline tests/testthat/test-rd-markdown.R]: They have to be words ;)

Comment (gaborcsardi) [inline tests/testthat/test-rd-markdown.R]: Oh, indeed. :)

Comment (krlmlr) [inline tests/testthat/test-rd-markdown.R]:
```r
roc_proc_text(rd_roclet(), "#' Title.\n#'\n#' @keywords Keywords **with bold**\nfoo <- function() {}")
```
$foo.Rd
% Generated by roxygen2: do not edit by hand
% Please edit documentation in Rtmpky86Q9/file53cac0b532b
\name{foo}
\alias{foo}
\title{Title.}
\usage{
foo()
}
\description{
Title.
}
\keyword{**with}
\keyword{Keywords}
\keyword{bold**}
```

Comment (krlmlr) [inline tests/testthat/test-rd-markdown.R]:
```r
roc_proc_text(rd_roclet(), "#' Title.\n#'\n#' @keywords Keywords \\strong{with bold}\nfoo <- function() {}")
```

```
$foo.Rd
% Generated by roxygen2: do not edit by hand
% Please edit documentation in Rtmpky86Q9/file53ca1d7558ed
\name{foo}
\alias{foo}
\title{Title.}
\usage{
foo()
}
\description{
Title.
}
\keyword{Keywords}
\keyword{\strong{with}
\keyword{bold}}
```

Comment (gaborcsardi) [inline tests/testthat/test-rd-markdown.R]: Oh, yeah, we don't, actually. So all is good. :)

**Comment by hadley** [comment]
Thanks!

**Pull request merged**